### ISSUE TITLE
Search Result OTF Screen Paths Incorrect

### ISSUE THREAD

**Issue by slominskir** [JeffersonLab/wmenu#9]
When attempting to follow a search result link to an On-The-Fly (OTF) generated EDM screen the URL path WMENU responds with is incorrect (so the screen does not load).    

For an example search for `FSD_NL17` and click the screen result that is found.

The WMENU code path in question is:
https://github.com/JeffersonLab/wmenu/blob/1a2e681c3cfcef15f9189a713f399924b3cfe5d7/src/main/webapp/resources/js/wmenu.js#L52

The code initializes a path part to `0` then looks in the search result JSON response for `menuData.actionId` and does not find it.  Other screen search results (non-OTF) this procedure works.

The expected result (originally, though now see https://github.com/JeffersonLab/wedm/issues/31) is:
https://epicsweb.jlab.org/wedm/screen?edl=/cs/opshome/edm/wedm/otf/FSDNodeScreen/FSDNodeScreen_FSD_NL17_

The actual result is that `/FSDNodeScreen/` path part is `/0/`.

The GET version of the search request and response in question can be found here: 
https://epicsweb.jlab.org/search/jmenu-cebaf/_search/?q=FSD_NL17&size=10&pretty=true

The actual search request uses a POST with parameters in the payload.  The payload of the POST request is:

```
size":10,"query":{"bool":{"must":[{"simple_query_string":{"query":"FSD_NL17","fields":["*","label^10","menuData.headings^2","menuData.labels^3","menuData.menu_labels^5","edlData.keywords^10","edlData.elements^3"],"default_operator":"AND"}},{"match":{"type":"ScreenAction"}}]}},"fields":["id","type","label","menuData.actionID","screen.value","screen.type"],"_source":false}
```

The server response:

```
{
    "took": 1,
    "timed_out": false,
    "_shards": {
        "total": 1,
        "successful": 1,
        "skipped": 0,
        "failed": 0
    },
    "hits": {
        "total": {
            "value": 2,
            "relation": "eq"
        },
        "max_score": 129.65942,
        "hits": [
            {
                "_index": "jmenu-cebaf-2",
                "_id": "FSDNodeScreen_FSD_NL17_",
                "_score": 129.65942,
                "fields": {
                    "screen.type": [
                        "otf"
                    ],
                    "id": [
                        "FSDNodeScreen_FSD_NL17_"
                    ],
                    "label": [
                        "FSD Node Screen - FSD_NL17 "
                    ],
                    "type": [
                        "ScreenAction"
                    ],
                    "screen.value": [
                        "OTFLauncher fsd fsdScreen -node  FSD_NL17"
                    ]
                }
            },
            {
                "_index": "jmenu-cebaf-2",
                "_id": "FSDNodeScreen_FSD_NL19_",
                "_score": 23.666798,
                "fields": {
                    "screen.type": [
                        "otf"
                    ],
                    "id": [
                        "FSDNodeScreen_FSD_NL19_"
                    ],
                    "label": [
                        "FSD Node Screen - FSD_NL19 "

                    ],
                    "type": [
                        "ScreenAction"
                    ],
                    "screen.value": [
                        "OTFLauncher fsd fsdScreen -node  FSD_NL19"
                    ]
                }
            }
        ]
    }
}
```

**Comment by slominskir** [comment]
**Note**: there are three ways to encounter an OTF screen on the web: 
1. JMENU directly linked from the menu
2. JMENU search result
3. WEDM screen may include a Command Widget to launch OTF Launcher

Theo is consolidating the pre-generated OTF screen path to be consistent and derive from a base64 encoded launcher command.  Both WMENU and WEDM will need to be upgraded.

**Comment by slominskir** [comment]
**Note**: new approach is dumping base64 encoded files into `edm/wedm/otf-pregen` dir.  Old `edm/wedm/otf` dir is now deprecated.    Looks like we're going to have three different conversion algorithms so hopefully they all share the same quirks (regex): 1. Menu and pregen server is PHP, WMENU is using client-side JavaScript, and WEDM likely will use server-side Java.

**Comment by theojlab** [comment]
Handful of example test cases here: []( https://github.com/JeffersonLab/otf-pregen/blob/main/tests/data/snippets1.edl)

**Comment by slominskir** [comment]
Fixed in [v1.5.0](https://github.com/JeffersonLab/wmenu/releases/tag/v1.5.0)
